lib/blacklight/models/answer.rb

```diff
@@ -0,0 +1,30 @@
+module Blacklight
+  class Answer
+    attr_reader :id
+    attr_accessor :fraction, :resp_ident, :feedback
+
+    def initialize(text, id = nil)
+      @answer_text = text
+      @resp_ident = ""
+      @fraction = ""
+      @feedback = ""
+      @points = 0
+      @id = id ? id : Blacklight.create_random_hex
```

Comment:
Another option:

```ruby
def initialize(text, id=Blacklight.create_random_hex)
   # ....
end
```

This is only effective if you're treating `id` solely as an optional parameter. If you're ever passing an actual `nil` value as the `id` parameter, and you want `nil` converted to something else, then do it like @bfcoder suggests instead.